### DISCUSSION TITLE
How to manage continuous dimension conversions

### DISCUSSION THREAD

**Discussion by tpadioleau** [Maison-de-la-Simulation/ddc#360] [Ideas]
In some application code a user wanted to make a conversion between continuous dimensions, something like:
```cpp
Coordinate<Vx, Vy> velocity;
Coordinate<T> time;
Coordinate<X, Y> coordinate = coordinate - time * velocity;
```
But to circumvent DDC, it was implemented as:
```cpp
Coordinate<X, Y> velocity;
double time;
Coordinate<X, Y> coordinate = coordinate - time * velocity;
```

**Comment by blegouix** [comment]
`operator*` should return `ElementType` when applied on two `TaggedVectors` ?

Alternatively, building new Tags with existing tags by making products and power elevations (in the sense of physical dimensions) could be awesome but looks complicated (need to ensure commutativity).

**Reply by tpadioleau** [reply]
Indeed it would make sense but I would advocate it is out of scope of DDC :/
